verl/trainer/ppo/ray_trainer.py

```diff
@@ -562,6 +562,34 @@ def _create_dataloader(self):
             self.config.actor_rollout_ref.actor.optim.total_training_steps = total_training_steps
             self.config.critic.optim.total_training_steps = total_training_steps
 
+    def _maybe_log_train_generations_to_wandb(self, inputs, outputs, rewards):
+        """Log a table of training samples to wandb."""
+        # Only log if wandb is used
+        if 'wandb' not in self.config.trainer.logger:
+            return
+        if hasattr(self.config.trainer, 'train_generations_to_log_to_wandb'):
+            generations_to_log = self.config.trainer.train_generations_to_log_to_wandb
+        else:
+            generations_to_log = -1
+
+        if generations_to_log == 0:
```

Comment:
What's the difference between paths 0 and -1?

Reply:
If it's 0 it doesn't log anything, if it's -1 it'd log all of the generations, if it's N it would log N generations